manifests/agent/config.pp

```diff
@@ -15,24 +11,28 @@
     value   => $puppet::agent::server,
   }
 
+  if $puppet::agent::ca_server {
```

Comment:
@tampakrap I think you can do:
```puppet
$real_ca_server = puppet::agent::ca_server ? {
  true    => $puppet::agent::ca_server,
  default => $puppet::agent::server,
}
```

To be conservative, it may be worth wrapping each of those in str2bool (from stdlib) in case somebody passes in a string "true". (e.g. `str2bool(true)`, str2bool(false)). Puppet 3.7 future parser is more sensitive to that sort of thing than older versions so I've started coding defensively around it.